packages/block-library/src/post-comments-form/index.php

```diff
@@ -46,9 +51,118 @@ function render_block_core_post_comments_form( $attributes, $content, $block ) {
 	// of the 'Reply' link that the user clicked by Core's `comment-reply.js` script.
 	$form = str_replace( 'class="comment-respond"', $wrapper_attributes, $form );
 
-	// Enqueue the comment-reply script.
-	wp_enqueue_script( 'comment-reply' );
+	$enhanced_submission = $block->context['enhancedSubmission'];
+
+	if ( $enhanced_submission ) {
+		$p = new WP_HTML_Tag_Processor( $form );
+		if ( $p->next_tag(
+			array(
+				'tag_name' => 'FORM',
+				'id'       => 'commentform',
+			)
+		) ) {
+			// Add the necessary directives.
+			$p->set_attribute( 'data-wp-on--submit', 'actions.core.comments.submit' );
+
+			if ( $p->next_tag( 'textarea' ) ) {
+				$p->set_attribute( 'data-wp-bind--value', 'context.core.comments.text' );
+				$p->set_attribute( 'data-wp-on--change', 'actions.core.comments.updateText' );
+			}
```

Comment:
We need to control the rest of the inputs (author, email…).

We can use the `name` attribute to find them because names should be stable among WordPress sites:
https://github.com/WordPress/wordpress-develop/blob/bf00a673a5f1d7f4707a6f914e1408526a8dc06a/src/wp-includes/comment.php#L3468-L3485

Something like this:

```php
while ( $p->next_tag( 'input' ) ) {
  if ( $p->get_attribute( 'name') === "..." )
}
```